common/persistence/dataInterfaces.go

```diff
@@ -396,6 +397,16 @@ type (
 		Version             int64
 	}
 
+	// ArchiveHistoryEventTask identifies a timer task for archival of history events of completed execution.
+	ArchiveHistoryEventTask struct {
+		VisibilityTimestamp time.Time
+		TaskID              int64
+		Version             int64
+		TargetDomainID      string
```

Comment:
You do not need to explicitly set DomainID, WorkflowID and RunID on timer tasks if they are created for same workflow execution.  TargetDomainID, TargetWorkflowID and TargetRunID are needed only for scenario where task is suppose to perform an action on another execution like transfer task for ChildWorkflowExecution processing.